docs/csharp/local-functions-vs-lambdas.md

```diff
@@ -26,46 +27,87 @@ Contrast that implementation with a version that uses lambda expressions:
 
 [!code-csharp[26_LambdaFactorial](../../samples/snippets/csharp/new-in-7/MathUtilities.cs#38_LambdaFactorial "Recursive factorial using lambda expressions")]
 
-First, lambda expressions are implemented by instantiating a delegate
-and invoking that delegate. Local functions are implemented as method calls.
-The instantiation necessary for lambda expressions means extra memory
-allocations, which may be a performance factor in time critical code paths.
-Local functions do not incur this overhead. In the example above, the local
-functions version has 2 fewer allocations than the lambda expression version.
-
-This recursive method is simple enough that the local function is implemented
-as a private method with a compiler generated name. Its only difference from
-other private methods is that it is semantically scoped inside the outer function.
-
-Second, local functions can be called before they are defined. Lambda
-expressions must be declared before they are defined. This
-means local functions are easier to use in recursive algorithms, as shown
-above.
-
-Notice that the version using the lambda expression must declare and initialize
-the lambda expression, `nthFactorial` before defining it. Not doing so results
-in a compile time error for referencing `nthFactorial` before assigning it.
+The local functions have names. The lambda expressions are anonymous methods
+that are assigned to variables that are `Func` or `Action` types. When you
+declare a local function the argument types and return type are part of the 
+function declaration. The argument types and return type are part of the
+variable type declaration when you declare a lambda expression instead of
+being part of the body of the lambda expression. Those two difference may
+result in more clear code.
+
+Second, local functions have different rules for definite assignment
+than lambda expressions. A local function declaration can be referenced
+from any code location where it is in scope. A lambda expression must be
+assigned before it can be accessed (or called through a delgate
+referencing the lambda expression.) Notice that the version using the
+lambda expression must declare and initialize the lambda expression,
+`nthFactorial` before defining it. Not doing so results in a compile
+time error for referencing `nthFactorial` before assigning it.
```

Comment:
The sentence of line 46 isn't related to anything here.

Reply:
I added text that ties them together better.